pkg/access_token/access_token_memory_test.go

```diff
@@ -16,84 +16,143 @@ func TestAccessTokenMemory(t *testing.T) {
 	config.StoreType = storageConfig.None
 	var tokenInMemory, _ = NewAccessTokenMemory()
 
+
 	Convey("Read", t, func() {
-		Convey("Returns true if Data output as expected", func() {
+
+		Convey("Given an access token with data saved in memory", func() {
 			dataActual := []byte{'t', 'e', 's', 't'}
 			tokenInMemory.Write(dataActual)
-			dataExpected, err := tokenInMemory.Read()
-			eq := reflect.DeepEqual(dataActual, dataExpected)
 
-			So(err, ShouldEqual, nil)
-			So(eq, ShouldEqual, true)
+			Convey("When running Read method", func() {
+				dataExpected, err := tokenInMemory.Read()
+
+				Convey("Finishes without raising an error", func() {
+					So(err, ShouldEqual, nil)
+				})
+
+				Convey("Returns the data the was written", func() {
+					eq := reflect.DeepEqual(dataActual, dataExpected)
+					So(eq, ShouldEqual, true)
+				})
+			})
 		})
 
-		Convey("Returns error if Data is nil", func() {
+		Convey("Given an access token's data is empty", func() {
 			tokenInMemory.Data = nil
-			_, err := tokenInMemory.Read()
 
-			So(err.Error(), ShouldEqual, "error reading access token, reason: data is empty")
+			Convey("Raises an error when reading the access token that the data is empty", func() {
+				_, err := tokenInMemory.Read()
+				So(err.Error(), ShouldEqual, "error reading access token, reason: data is empty")
+			})
 		})
 	})
 
 	Convey("Write", t, func() {
-		Convey("Returns no error if Data input is not nil", func() {
+
+		Convey("Given an access token with data", func() {
 			dataActual := []byte{'t', 'e', 's', 't'}
-			err := tokenInMemory.Write(dataActual)
 
-			So(err, ShouldEqual, nil)
+			Convey("Writes the access token to memory without raising an error", func() {
+				err := tokenInMemory.Write(dataActual)
+				So(err, ShouldEqual, nil)
+			})
+
+			Convey("When running Read method", func() {
```

Comment:
this should go inside the convey above. also, please see my comment on this in `access_token_file`

Reply:
(read with `os`, not `tokenInFile`